forgevtt-module.js

```diff
@@ -1985,27 +2029,87 @@ class ForgeVTT_FilePicker extends FilePicker {
             }
         }
         // Sort the share-URL buckets by name
-        buckets.push(...sharedBuckets.sort((a, b) => a.label.localeCompare(b.label)));
+        sharedBuckets.sort((a, b) => a.label.localeCompare(b.label));
+        buckets.push(...sharedBuckets);
+        if (ForgeAPI.lastStatus) {
+            this._forgeBucketIndex = buckets;
+        }
         return buckets;
     }
 
+    /**
+     * Retrieves the Forge VTT bucket based on the provided key or index.
+     * @param {string|number} bucketKey - The key or index of the bucket to retrieve.
+     * @returns {object} The Forge VTT bucket object.
+     */
+    static _getForgeVttBucket(bucketKey) {
+        const buckets = this._getForgeVTTBuckets();
+        // From Foundry v12, buckets are keyed by index not by hash
+        const isHashKey = isNaN(bucketKey) || !foundry.utils.isNewerVersion(ForgeVTT.foundryVersion, "12");
+        const bucketIndex = isHashKey ? buckets.findIndex((b) => b.key === bucketKey) : bucketKey;
+        return buckets[bucketIndex];
+    }
+
+    /**
+     * Get the bucket key (relative to Foundry version) for the specified bucket.
+     *
+     * @param {Object} bucket - The bucket object.
+     * @returns {string|number} - The bucket key.
+     */
+    _getBucketKey(bucket) {
+        const buckets = this.constructor._getForgeVTTBuckets();
+        return foundry.utils.isNewerVersion(ForgeVTT.foundryVersion, "12")
+            ? buckets.findIndex((b) => b.key === bucket.key)
+            : bucket.key;
+    }
```

Comment:
This is effectively the fix for the upload: v12 uses "numeric" indices rather than the hashKey to reference the active bucket.